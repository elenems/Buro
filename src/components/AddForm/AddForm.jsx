import React, { useState } from "react";
import TextField from "@material-ui/core/TextField";
import InputLabel from "@material-ui/core/InputLabel";
import FormControl from "@material-ui/core/FormControl";
import Select from "@material-ui/core/Select";
import Box from "@material-ui/core/Box";
import Typography from "@material-ui/core/Typography";
import Button from "@material-ui/core/Button";

export default function AddForm() {
  const [title, setTitle] = useState("");
  const [location, setLocation] = useState("");
  const [description, setDescription] = useState("");
  const handleChange = e => {
    const id = e.target.id;
    const value = e.target.value;
    switch (id) {
      case "title":
        setTitle(value);
        break;
      case "location":
        setLocation(value);
        break;
      case "description":
        setDescription(value);
        break;
    }
  };

  const send = () => {

  }
  return (
    <form>
      <Typography component="h2" variant="h4">
        Розмістіть знайдену річ
      </Typography>
      <Box
        minHeight="360px"
        justifyContent="space-between"
        display="flex"
        flexDirection="column"
      >
        <TextField
          id="title"
          label="Заголовок"
          value={title}
          onChange={handleChange}
          margin="normal"
        />
        <FormControl>
          <InputLabel htmlFor="location">Місто</InputLabel>
          <Select
            native
            value={location}
            onChange={handleChange}
            inputProps={{
              name: "location",
              id: "location"
            }}
          >
            <option value="Lviv">Львів</option>
            <option value="Kyiv">Київ</option>
            <option value="Kharkiv">Харків</option>
          </Select>
        </FormControl>
        <textarea value={description} onChange={handleChange} placeholder="Опис" id="description"></textarea>
        <Button onClick={send} color="primary">
          Відправити
        </Button>
      </Box>
    </form>
  );
}
