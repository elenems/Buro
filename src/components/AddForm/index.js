import AddForm from "./AddForm.jsx";
export default AddForm;
