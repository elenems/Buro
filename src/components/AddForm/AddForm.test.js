import React from "react";
import { shallow, mount } from "enzyme";
import AddForm from "../AddForm";
import Typography from "@material-ui/core/Typography";
import Button from "@material-ui/core/Button";
import TextField from "@material-ui/core/TextField";
import Select from "@material-ui/core/Select";
import { wrap } from "module";

describe("AddForm", () => {
  it("Matches snapshot", () => {
    const wrapper = shallow(<AddForm />).debug();
    expect(wrapper).toMatchSnapshot();
  });

  it("Renders error text on error", () => {
    const wrapper = shallow(<AddForm />);
    wrapper.find(Button).simulate("click", { preventDefault: () => {} });
    expect(
      wrapper.findWhere(
        n => n.text() === "Завантажте фото" && n.prop("color") === "error"
      ).length
    ).toBe(1);
  });

  it("Renders message on success", () => {
    const wrapper = shallow(<AddForm />);
    wrapper.update();
    wrapper
      .findWhere(n => n.prop("type") === "file")
      .simulate("change", {
        target: {
          files: [{ file: {} }]
        }
      });
    wrapper.find(Button).simulate("click", { preventDefault: () => {} });
  });

  it("Sets new title", () => {
    const wrapper = shallow(<AddForm />);
    wrapper.find(TextField).simulate("change", {
      target: {
        id: "title",
        value: "New title"
      }
    });
    expect(wrapper.find(TextField).prop('value')).toBe('New title');
  });

  it("Sets new lcoation", ()=>{
    const wrapper = shallow(<AddForm />);
    wrapper.find(Select).simulate('change',{target: {
        id: "location",
        value: "Kharkiv"
      }})
    expect(wrapper.find(Select).prop('value')).toBe("Kharkiv")
  })

  it("Sets new description", ()=>{
    const wrapper = shallow(<AddForm />);
     wrapper.find('#description').simulate('change',{
        target: {
            id: "description",
            value: "New description"
          }
     })
     expect(wrapper.find('#description').prop('value')).toBe('New description');
  })
});
