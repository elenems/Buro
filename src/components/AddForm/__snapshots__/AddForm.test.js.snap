// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddForm Matches snapshot 1`] = `
"<form>
  <WithStyles(ForwardRef(Typography)) component=\\"h2\\" variant=\\"h4\\">
    Розмістіть знайдену річ
  </WithStyles(ForwardRef(Typography))>
  <Styled(MuiBox) minHeight=\\"360px\\" justifyContent=\\"space-between\\" display=\\"flex\\" flexDirection=\\"column\\">
    <WithStyles(ForwardRef(TextField)) id=\\"title\\" label=\\"Заголовок\\" value=\\"\\" onChange={[Function: handleChange]} margin=\\"normal\\" />
    <WithStyles(ForwardRef(FormControl))>
      <WithStyles(ForwardRef(InputLabel)) htmlFor=\\"location\\">
        Місто
      </WithStyles(ForwardRef(InputLabel))>
      <WithStyles(ForwardRef(Select)) native={true} value=\\"Lviv\\" onChange={[Function: handleChange]} inputProps={{...}}>
        <option value=\\"Lviv\\">
          Львів
        </option>
        <option value=\\"Kyiv\\">
          Київ
        </option>
        <option value=\\"Kharkiv\\">
          Харків
        </option>
      </WithStyles(ForwardRef(Select))>
    </WithStyles(ForwardRef(FormControl))>
    <textarea value=\\"\\" onChange={[Function: handleChange]} placeholder=\\"Опис\\" id=\\"description\\" />
    <span className=\\"label\\">
      Фото
    </span>
    <input type=\\"file\\" name=\\"file\\" onChange={[Function: onChangeHandler]} />
    <WithStyles(ForwardRef(Button)) onClick={[Function: send]} color=\\"primary\\">
      Відправити
    </WithStyles(ForwardRef(Button))>
  </Styled(MuiBox)>
</form>"
`;
